html/docs/design/sync-ot.html

```diff
@@ -0,0 +1,435 @@
+            </p>
+
+            <h4 id="operational-transforms">Operational Transforms</h4>
+
+            <p>
+            Synchronization is based on <a href="https://en.wikipedia.org/wiki/Operational_transformation">operational transformation</a>.
+            This section will assume some familiarity with the concept.
+            </p>
+
+            <h4 id="state-and-operations">State and Operations</h4>
+
+            <p>
+            At a given time, the set of tasks in a replica's storage is the essential <em>state</em> of that replica.
+            This state has the form of an unordered set of tasks indexed by UUID.
+            Each task is a collection of key/value pairs, both represented as strings.
+            <em>Any</em> state of this form is valid from the perspective of the synchronization model.
+            </p>
+
+            <p>
+            All modifications to that state occur via <em>operations</em>.
+            We can draw a network, or graph, with the nodes representing states and the edges representing operations.
+            For example:
+            </p>
+<pre>  o -- State: {"abc-123": {"description": "get groceries", "priority": "L"}}
+  |
+  | -- Operation: set abc-123 "priority" to "H"
+  |
+  o -- State: {"abc-123": {"description": "get groceries", "priority: "H"}
+</pre>
+
+            <p>
+            Operations in Taskwarrior have one of three forms:
+              <ul>
+                  <li><code>Create(uuid)</code> - create a new task with the given UUID.</li>
+                  <li><code>Update(uuid, key, value)</code> - update the given task with the given key and value, where value may be null to indicate removal of a key.</li>
+                  <li><code>Delete(uuid)</code> - delete the given task.</li>
+              </ul>
+            </p>
+
+            <p>
+            For those familiar with distributed version control systems, a state is analogous to a revision, while an operation is analogous to a commit.
+            </p>
+
+            <p>
+            Fundamentally, synchronization involves all replicas agreeing on a single, linear sequence of operations and the state that those operations create.
+            Since the replicas are not connected, each may have additional operations that have been applied locally, but which have not yet been agreed on.
+            The synchronization process uses operational transformation to "linearize" those operations.
+            This process is analogous (vaguely) to rebasing a sequence of Git commits.
+            </p>
+
+            <h4 id="versions">Versions</h4>
+
+            <p>
+            Occasionally, database states are given a name (that takes the form of a UUID).
+            The system as a whole (all replicas) constructs a branch-free sequence of versions and the operations that separate each version from the next.
+            The version with the nil UUID is implicitly the empty database.
```

Comment:
Perhaps we should use a different term here, to avoid overloading of UUIDs? :thinking:

Reply:
I see this is referred to as _version_ in latter parts of the text.

Reply:
The term is "version", yes -- the uuid is the format (although it doesn't have to be)